src/contour/TerminalSession.h

```diff
@@ -166,6 +168,7 @@ class TerminalSession: public terminal::Terminal::Events
     std::string profileName_;
     config::TerminalProfile profile_;
     std::string programPath_;
+    Controller *controller_;
```

Comment:
```suggestion
    Controller& controller_;
```

I though highly recommend that to be a reference instead to give the guarantee that this will never be NULL.